src/ripple/rpc/impl/KeypairForSignature.cpp

```diff
@@ -0,0 +1,47 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2015 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/rpc/impl/KeypairForSignature.h>
+
+namespace ripple {
+namespace RPC {
+
+SignatureKeypair KeypairForSignature (Json::Value const& params)
```

Comment:
Shrug.  When I saw `KeypairForSignature` I thought it might be doing some recovery algorithm or something.  That was immediately followed up by "oh hrmm, it's not going to recover a secret, you silly duffer".  Made sense after looking again.  But still, it's a function with struct casing? I'm partial to `derive_keypair` and `SigningKeyPair` or just `KeyPair`.   Maybe `SignatureEditionKeypair` (or maybe not :)